### ISSUE TITLE
Make use of eps for numerical stability more consistent

### ISSUE THREAD

**Issue by albanD** [pytorch/pytorch#50473]
Different ops have `eps` as argument to ensure numerical stability when dividing by Tensors that can be 0.
In general, we should handle these epsilon consistently as discussed in https://github.com/pytorch/pytorch/pull/49807#discussion_r554097359

While some paper explicitly say that we should use `val + eps`  (like batchnorm and other normalizations).
We should fix the other ones to use clamping to avoid unnecessary changes of the values far from 0.

In particular, we should check the following functions:
- cosine similarity
- norms
- cdist
